README.md

```diff
@@ -0,0 +1,80 @@
+# Rust libraries for the Qubes OS GUI Protocol
+
+This provides Rust libraries for the Qubes OS GUI Protocol.  While the existing
+[agent] and [daemon] have served Qubes OS users well, they are monolithic C
+codebases and are not suitable for use as libraries.  Additionally, they are
+very much tied to the X Window System, whereas the future of FLOSS graphics is
+Wayland.  Finally, the Qubes OS GUI protocol is underspecified, so these
+libraries aim to provide a definition of the protocol that is independent of any
+particular implementation.
+
+## Organization
+
+To maximize portability and build-time parallelism, and to prevent circular
+dependencies, these libraries are broken up into several different crates.  Each
+crate serves a specific purpose, and some are reusable outside of Qubes OS.
+
+
+### Qubes-Castable
+
+qubes-castable is a core crate that provides support for _castable_ structs ―
+that is, structs that can safely be converted to a byte slice.  It is not
+specific to Qubes OS in any way.  Unlike the rest of the code, this crate is
+dual-licensed under the MIT License and the Apache License, Version 2.0.  This
+is the same license used by the Rust Programming Language itself.  It is
+`#[no_std]` and depends only on libcore, so it can be used anywhere.
```

Comment:
This sounds very similar to https://lib.rs/crates/zerocopy

I've been told that 3rd-party code is being avoided because the crates distributed through crates.io aren't signed. However code hashes get stored in `Cargo.lock`, and vendoring the dependencies into the repository also isn't too problematic. If we can benefit from the battle-tested code of established crates in the Rust ecosystem, I don't think writing our own `unsafe` primitives is reasonable.

If we decide against that, I'll take a close look at `qubes-castable`.